Userland/Libraries/LibAudio/WavLoader.cpp

```diff
@@ -191,22 +187,19 @@ MaybeLoaderError WavLoaderPlugin::parse_header()
     size_t bytes_read = 0;
 
     auto read_u8 = [&]() -> ErrorOr<u8, LoaderError> {
-        u8 value;
-        LOADER_TRY(m_stream->read(Bytes { &value, 1 }));
+        u8 value = LOADER_TRY(m_stream->read_value<u8>());
         bytes_read += 1;
         return value;
     };
 
     auto read_u16 = [&]() -> ErrorOr<u16, LoaderError> {
-        u16 value;
-        LOADER_TRY(m_stream->read(Bytes { &value, 2 }));
+        u16 value = LOADER_TRY(m_stream->read_value<u16>());
         bytes_read += 2;
         return value;
     };
 
     auto read_u32 = [&]() -> ErrorOr<u32, LoaderError> {
-        u32 value;
-        LOADER_TRY(m_stream->read(Bytes { &value, 4 }));
+        u32 value = LOADER_TRY(m_stream->read_value<u32>());
```

Comment:
I'm just noticing that this will all horribly die on non-little endian machines... can you quickly insert a AK::convert_between_host_and_little endian?

Reply:
Fixed, but wouldn't we also have to convert the hardcoded magic values to always be little endian then?

Reply:
Nope, since x86 is already little endian, so this code worked before only because we are little endian, and it will now also work on big endian. I.e. the actual values read back are still the same on an x86 build.